ical.py

```diff
@@ -2,42 +2,47 @@
 
 import asyncio
 import os
-from spond import spond
+
 from ics import Calendar, Event
-from config import username, password
 
-if not os.path.exists('./exports'):
-    os.makedirs('./exports')
+from config import password, username
+from spond import spond
+
+if not os.path.exists("./exports"):
+    os.makedirs("./exports")
 
 ics_file = os.path.join("./exports", "spond.ics")
 
+
 async def main():
     s = spond.Spond(username=username, password=password)
     c = Calendar()
-    c.method = 'PUBLISH'
+    c.method = "PUBLISH"
     events = await s.get_events()
     for event in events:
         e = Event()
-        e.uid = event['id']
-        e.name = event['heading']
-        e.description = event['description']
-        e.begin = event['startTimestamp']
-        e.end = event['endTimestamp']
-        e.sequence = event['updated']
-        if 'cancelled' in event and event['cancelled']:
-            e.status = 'Cancelled'
-        if 'location' in event:
-            e.location = "{}, {}".format(event['location']['feature'], event['location']['address'])
+        e.uid = event["id"]
+        e.name = event["heading"]
+        e.description = event["description"]
+        e.begin = event["startTimestamp"]
+        e.end = event["endTimestamp"]
+        e.sequence = event["updated"]
+        if "cancelled" in event and event["cancelled"]:
+            e.status = "Cancelled"
+        if "location" in event:
+            e.location = "{}, {}".format(
+                event["location"]["feature"], event["location"]["address"]
+            )
         c.events.add(e)
-    with open(ics_file, 'w') as out_file:
+    with open(ics_file, "w") as out_file:
         out_file.writelines(c)
     await s.clientsession.close()
 
+
 loop = asyncio.new_event_loop()
 asyncio.set_event_loop(loop)
 asyncio.run(main())
 
 
 # loop = asyncio.get_event_loop()
```

Comment:
These can be removed if the "new" way to run async works in 3.7

Reply:
This code isn't changed by this pull request, so have raised a new issue https://github.com/Olen/Spond/issues/47